src/test/java/org/candlepin/resource/UserResourceTest.java

```diff
@@ -97,7 +97,8 @@ public void testCreateUser() {
 
         assertNotNull(existing);
         assertEquals(dto.getUsername(), existing.getUsername());
-        assertEquals(Util.hash(dto.getPassword()), existing.getHashedPassword());
+        assertTrue(
+            OpenBSDBCrypt.checkPassword(existing.getPassword(), dto.getPassword().toCharArray()));
```

Comment:
This is an abstraction leak. Tests for the UserResource should not have to know implementation details for a completely unrelated object. This is one of those cases where we could mock out the UserServiceAdapter to avoid this problem (either as an actual mock or with a customized implementation for testing).

Reply:
I mocked the adapter and I modified the tests to test only UserResource, some things were tested in more places in our codebase and I added tests that were missing and I left it in the new commit in case we didn't like the changes